tests/e2e/main_test.go

```diff
@@ -77,3 +82,49 @@ func TestLintMetrics(t *testing.T) {
 		t.Fatalf("the problems encountered in Lint are: %v", problems)
 	}
 }
+
+func TestDefaultCollectorMetricsAvailable(t *testing.T) {
+	buf := &bytes.Buffer{}
+
+	err := framework.KsmClient.metrics(buf)
+	if err != nil {
+		t.Fatalf("failed to get metrics from kube-state-metrics: %v", err)
+	}
+
+	resources := map[string]struct{}{}
+	files, err := ioutil.ReadDir("../../internal/store/")
```

Comment:
Maybe instead of this we could use this list? To avoid the reading of directories.

https://github.com/kubernetes/kube-state-metrics/blob/0bc19488277fb94adbfc9b6f64611ccd4cea1e78/pkg/options/collector.go#L28-L55

Reply:
Actually I tried at first :0

But it could not work. Because the resource name in this list is plural form and that in metric name is singular. Also the resource name is not exactly corresponding.

If we want to use this list directly, we have to change the command line config as well which is a break change:) So I still use the name of directories.

Reply:
I think the main Kubernetes repository maintains a list of the resource names along with its plurals. We could, perhaps, leverage that. WDYT @YaoZengzeng?

Reply:
@tariq1890 I found this [ResourceSingularizer](https://github.com/kubernetes/apimachinery/blob/v0.17.3/pkg/api/meta/restmapper.go#L149) function, but it needs a `DefaultRESTMapper` that has already been initialized with all possible resource kinds, so I don't think this could be used. Did you have another list in mind?